code-quality/validation/README.md

```diff
@@ -0,0 +1,91 @@
+<!-- Last updated 9-25-19 -->
+
+# Input Validation 
+
+Input validation is is your first line of defense when creating a secure application.  
+
+## Whitelist or Blacklist? 
+
+The first concept of good input validation is whitelisting versus blacklisting. Whitelist, or inclusive validation, defines a set of valid characters while blacklist, or exclusive validation, defines a set of invalid characters to try to remove. 
+
+Input validation using blacklisting is generally more challenging to implement so it is better practice to perform validation using whitelisting. A great way of defining a whitelist for input validation is to leverage Regular Expressions(RegEx). 
+
+## RegEx Password Validation Examples (JavaScript)
+
+- Mininum length of any 6 chars but newline: 
+```javascript
+/^.{6,}$/
+```
+
+- Check for at least one uppercase character: 
+```javascript
+/([A-Z]+)/g
+```
+
+- Check for at least one lowercase character:
+```javascript
+/([a-z]+)/g
+```
+
+- Check for at least one numeric character: 
+```javascript
+/([0-9]+)/g
+```
+
+
+- Check for at least one special character:
+```javascript
+/([!"#$%&'()*+,\-./:;<=>?@[\]^_`{|}~])+/g
```

Comment:
I don't think we should include -, <, or ', as these can be used in SQL Injection and similar attacks.

Reply:
We should be sanitizing/encoding the data we get in the server, so the SQL injection shouldn't be a thing.

We also shouldn't prevent the user from using them if it makes the password stronger.